exonum/src/blockchain/mod.rs

```diff
@@ -541,6 +546,25 @@ impl Blockchain {
     }
 }
 
+fn service_execute(service: &Service, mut fork: &mut Fork) {
+    fork.checkpoint();
+    match panic::catch_unwind(panic::AssertUnwindSafe(|| service.execute(&mut fork))) {
```

Comment:
Continuing @vitvakatu's question: what if `fork` is passed here instead of `&mut fork` (seems kind of redundant)?